lhotse/recipes/gigaspeech.py

```diff
@@ -1,41 +1,79 @@
+        if maybe_manifests is not None:
+            return maybe_manifests
+
     manifests = defaultdict(dict)
-    for subset in subsets:
-        recordings = []
-        segments = []
-        for audio in gigaspeech.audios(subset):
-            recordings.append(Recording.from_file(gigaspeech.root_path / audio['path'],
-                                                  recording_id=audio['aid']))
-            for seg in audio['segments']:
-                segments.append(SupervisionSegment(id=seg['sid'],
-                                                   recording_id=audio['aid'],
-                                                   start=seg['begin_time'],
-                                                   duration=seg['end_time'] - seg['begin_time'],
-                                                   channel=0,
-                                                   language='English',
-                                                   speaker=seg['speaker'],
-                                                   text=seg['text_tn']))
-        manifests[subset] = {
-            'recordings': RecordingSet.from_recordings(recordings),
-            'supervisions': SupervisionSet.from_segments(segments)
-        }
+    with ThreadPoolExecutor(num_jobs) as ex:
+        for part in subsets:
+            futures = []
+            for audio in tqdm(gigaspeech.audios(part), desc='Distributing tasks', leave=False):
+                futures.append(ex.submit(parse_utterance, audio, gigaspeech.root_path))
+
+            recordings = []
+            supervisions = []
+            for future in tqdm(futures, desc='Processing', leave=False):
+                result = future.result()
+                if result is None:
+                    continue
+                recording, segments = result
+                recordings.append(recording)
+                supervisions += segments
+
+            manifests[part] = {
+                'recordings': RecordingSet.from_recordings(recordings),
+                'supervisions': SupervisionSet.from_segments(supervisions)
+            }
+
+            if output_dir is not None:
+                manifests[part]['recordings'].to_json(output_dir / f'recordings_{part}.json')
+                manifests[part]['supervisions'].to_json(output_dir / f'supervisions_{part}.json')
+
     return dict(manifests)
+
+
+def parse_utterance(
+        audio: Any,
+        root_path: Path
+) -> Optional[Tuple[Recording, List[SupervisionSegment]]]:
+    recording = Recording.from_file(root_path / audio['path'], recording_id=audio['aid'])
+    segments = []
+    for seg in audio['segments']:
+        segments.append(SupervisionSegment(id=seg['sid'],
+                                           recording_id=audio['aid'],
+                                           start=Seconds(seg['begin_time']),
+                                           duration=Seconds(seg['end_time'] - seg['begin_time']),
```

Comment:
you might want to round the duration with sth like `round(…, ndigits=8)`, otherwise there will be entries like 17.0000000002